src/components/assets/YourProject.vue

```diff
@@ -35,11 +44,40 @@ export default defineComponent({
       required: true,
     },
   },
-  setup() {
+  setup(props) {
     useClaimAll();
     const { navigateOwnerPage } = useDappStakingNavigation();
+    const { rewards } = useDappStaking();
+    const dAppRewardsMap = ref<{ dapp: string, rewards: bigint }[]>([]);
+
+    const checkIsRewards = (address: string): boolean => {
+      const dapp = dAppRewardsMap.value.find((it) => it.dapp === address);
+      if (dapp) {
+        return dapp.rewards > 0;
+      }
+      return false;
+    }
+
+    const setDappRewardsMap = async (): Promise<void> => {
+      const dAppRewardsArray: { dapp: string, rewards: bigint }[] = [];
+      if (props.ownDapps.length === 0) return
+      const stakingV3service = container.get<() => IDappStakingService>(
```

Comment:
You can remove this if you use `getDappRewards`, from `useDappStaking`. My general idea was to move all interaction with services to the hook.